File: scripts/check-today.py
from supabase import create_client
import os
import sys
sys.path.append('C:\\Projects\\dayflow-scheduler')

os.environ['SUPABASE_URL'] = 'https://prloxvewcsxaptzgxvyy.supabase.co'
os.environ['SUPABASE_SERVICE_KEY'] = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InBybG94dmV3Y3N4YXB0emd4dnl5Iiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTcyNjkxODU4MCwiZXhwIjoyMDQyNDk0NTgwfQ.IH4e6IB-S2rp_AwT0xH2YAJ7l7aZ4HU0YIJcJhySxCc'

sb = create_client(os.environ['SUPABASE_URL'], os.environ['SUPABASE_SERVICE_KEY'])

print("=== TODAY'S TASKS (2025-11-16) ===")
tasks = sb.table('scheduled_tasks').select('title,is_completed,is_deleted,start_time,template_id').eq('local_date', '2025-11-16').execute()
for t in tasks.data:
    time_str = str(t.get('start_time', 'None'))[:16] if t.get('start_time') else 'None'
    print(f"{t['title']:30} comp={t['is_completed']} del={t['is_deleted']} time={time_str}")

print("\n=== YESTERDAY'S TASKS (2025-11-15) ===")
yesterday = sb.table('scheduled_tasks').select('title,is_completed,is_deleted').eq('local_date', '2025-11-15').in_('title', ['Recycle', 'Black bags', 'Spraying', 'Clear grotto', 'Clear leaves']).execute()
for t in yesterday.data:
    print(f"{t['title']:30} comp={t['is_completed']} del={t['is_deleted']}")

File: scripts/check-today.mjs
import { createClient } from '@supabase/supabase-js';
import * as dotenv from 'dotenv';

dotenv.config({ path: '.env.local' });

const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL;
const supabaseKey = process.env.SUPABASE_SERVICE_ROLE_KEY;

if (!supabaseUrl || !supabaseKey) {
  console.error('Missing environment variables');
  process.exit(1);
}

const sb = createClient(supabaseUrl, supabaseKey);

const userId = '3c877140-9539-47b9-898a-45eeab392e39';

const { data, error } = await sb
  .from('scheduled_tasks')
  .select('title, start_time, end_time, is_routine, is_deleted, is_completed')
  .eq('user_id', userId)
  .eq('local_date', '2025-11-29')
  .order('start_time');

if (error) {
  console.error('Error:', error);
  process.exit(1);
}

console.log('Today\'s schedule (ALL tasks including deleted):');
data.forEach(task => {
  const start = new Date(task.start_time);
  const end = new Date(task.end_time);
  const routine = task.is_routine ? ' (routine)' : '';
  const flags = ` del:${task.is_deleted} done:${task.is_completed}`;
  console.log(`${start.toLocaleTimeString('en-GB', {hour: '2-digit', minute: '2-digit'})} - ${end.toLocaleTimeString('en-GB', {hour: '2-digit', minute: '2-digit'})} ${task.title}${routine}${flags}`);
});
